packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,157 @@
+
+    @observable
+    public currentDisableResize = false;
+
+    ...
+}
+```
+Note that these properties are not attributes, and thus are not set by clients on the components via markup.
+
+These properties are expected to be updated by both the initial state of separate, column-specific APIs and interactive operations that the `Table` would manage.
+
+#### `currentFractionalWidth` vs `currentFixedWidth` behavior
+
+- The values supplied to the `currentFractionalWidth` property have the same meaning as the values supplied with the [`fr` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#the_fr_unit) for the CSS `grid-template-columns` property.
+- The values supplied to the `currentFixedWidth` property are meant to be pixel-based values.
+- The table will use `currentFixedWidth` over `currentFractionalWidth` when both are set.
+
+#### `currentMinWidth` behavior
+
+Open Question: Is there a minimum width considered too small? If a user is allowed to provide a mimimum width smaller than the default, how should the various components within the header behave? If a user attempts to set a minimum width to an unsupported value, what is the appropriate response (i.e. should we simply update the `TableValidity` state?)?
+
+#### **Mixin Example**
+
+We can help facilitate proper implementation for concrete column types by providing mixins that define expected attribute APIs:
+
+```
+export abstract class FractionalWidthColumn extends TableColumn {
+    @attr({ attribute: 'fractional-width', converter: nullableNumberConverter })
+    public fractionalWidth = 1;
+
+    @attr({ attribute: 'disable-resize' })
+    public disableResize = false;
+
+    @attr({ attribute: 'min-width' })
+    public minWidth?: number;
+
+    public fractionalWidthChanged(): void {
+        this.currentFractionalWidth = this.fractionalWidth;
+    }
+
+    public disableResizeChanged(): void {
+        this.currentDisableResize = this.disableResize;
+    }
+
+    public minWidthChanged(): void {
+        this.currentMinWidth = this.minWidth;
+    }
+}
+
+```
+_Note: We do not need to provide a `maxWidth` as it seems like an unnecessarily limiting API._
+
+These mixin classes must extend `TableColumn` as that class is what defines the APIs that the internal components will rely on to create the appropriate layout state (and thus are updated by the mixin implementation as shown above). Additionally, they must be marked as `abstract` in order to avoid the requirement of implementing the abstract APIs of `TableColumn`.
+
+We can then apply that mixin to a concrete TableColumn class (e.g. `TableColumnText`) like so:
+```
+export class TableColumnText extends TableColumn<...> {
+    ...
+}
+
+applyMixins(TableColumnText, FractionalWidthColumn);
+```
+
+The `applyMixins` method is a helper function provided by FAST, that we leverage in other places in Nimble already.
```

Comment:
Might be worth a note explaining why we're proposing a mixin pattern rather than a base class pattern. I'm guessing it's because the resize strategy isn't "fundamental" enough to merit making each column *be* a particular resize strategy?

Mixins might also be worth documenting as a pattern in one of our CONTRIBUTING docs.

Reply:
Added a note on the matter. Regarding documenting the pattern in the CONTRIBUTING docs, I would prefer to handle that separately so I can give it more focused thought as to what the content of those docs should be.